lib/jsanalyze.js

```diff
@@ -181,16 +186,52 @@ exports.analyzeJs = function analyzeJs(contents, opts) {
 			comments: false
 		});
 
-		options.presets.push([ babili, {
+		options.presets.push([ minify, {
 			mangle: false,
 			deadcode: false
 		} ]);
 
-		options.plugins.push(require.resolve('babel-plugin-transform-property-literals'));
+		options.plugins.push(require.resolve('@babel/plugin-transform-property-literals'));
 	}
 
 	if (options.presets.length) {
-		// FIXME we can't re-use the ast here, because we traversed it
+
+		// rollup is required to workaround circular references unsupported by JavascriptCore
+		// this should only run on 'app.js', bundling all dependencies into one file
+		if (opts.rollup) {
```

Comment:
I think the ability to include rollup in the pipeline is interesting and useful, but seems like a huge new addition here that we need to test/try out a bit before I'd feel too comfortable.

I assume the driver here was using ES6 imports with circular references on iOS?

Rollup should only be run on the single app entry point, and would be akin to minification in my mind.

I think we're quickly bumping into a large architectural issue here:
- We need to find a way to be able to run some arbitrary set of transformations on the app source code
- in a way that doesn't clash with one another
- with the output of one transformation being the input to the next (i.e the modified source and source map).